Push notifications ios

jcesarmobile opened the pull request:
Moved part of the LocalNotifications code to a common class as both Push and Local are handled by UNUserNotificationCenter

mlynch commented on ios/Capacitor/Capacitor/CAPUNUserNotificationCenterDelegate.swift:
```diff
@@ -0,0 +1,145 @@
+    // Override point for customization after application launch.
+    let center = UNUserNotificationCenter.current()
+    center.requestAuthorization(options:[.badge, .alert, .sound]) { (granted, error) in
+      // Enable or disable features based on authorization.
```
Something supposed to be here?

mlynch commented on ios/Capacitor/Capacitor/Plugins/PushNotifications.swift:
```diff
@@ -1 +1,83 @@
+    }
+    notifyListeners("didFailToRegisterForRemoteNotificationsWithError", data:[
+      "error": error.localizedDescription
+      ])
```
A few of these dangling `])` at the wrong indentation level

jcesarmobile commented:
Fixed that and removed some print lines I left behind.

pushNotificationReceived is fired when the notification is received and the app is in foreground, or when you open the app by tapping a notification from notification center.

mlynch commented:
Saweeeet!
